### ISSUE TITLE
(1.3.0 Preview Build) Bitrate is way higher than what I select!

### ISSUE THREAD

**Issue by razorlikes** [Xaymar/OBS-AMD-Advanced-Media-Framework#39]
Settings: http://prntscr.com/c8kx2e
Video info: http://prntscr.com/c8kwox

I think those two pictures speak for themselves. Also there still is some kind of flickering throughout the whole video, not as much as there was with the broken B-Frames, but still a little. I'll see if I can create an example where it is easily visible.

I tried to experiment with my settings a bit but the issue persists.

Also, is the advanced encoder gone for now?

**Comment by Xaymar** [comment]
Update to the latest uploaded build, I fixed a typo in the Set*Bitrate commands. I've not experienced the flickering issue myself.

Advanced Encoder will return once I've done everything necessary.

**Comment by razorlikes** [comment]
I'm already using the latest build! Tried to install it using the .zip file as well as the installer.

**Comment by Xaymar** [comment]
Could you upload the last log file? I'll take a look then.

**Comment by razorlikes** [comment]
There you go!
[2016-08-21 22-05-41.txt](https://github.com/Xaymar/OBS-AMD-Media-Framework/files/429079/2016-08-21.22-05-41.txt)

**Comment by Xaymar** [comment]
Hmm, 1950kbit is a little low for that resolution and framerate. AMF is most likely not capable of going that low yet. You'll have to open an issue on the [AMF SDK](https://github.com/GPUOpen-LibrariesAndSDKs/AMF) Issue Tracker, I'm afraid.

**Comment by razorlikes** [comment]
Wait, is it using 1080p? I set it to 960x540! It should be able to handle that.

**Comment by Xaymar** [comment]
No it's using 960x540. But this has been an issue with hardware encoders for a long time, bitrates lower than 2500kbit are near impossible to consistently achieve. You might have better luck at 30 fps.

**Comment by wazerstar** [comment]
Tested with the latest 1.3pre? version 

http://prntscr.com/c8pynh
http://prnt.sc/c8pyx2

It jumped from 16xxx to 30xxx

Took a chance to try stream too, dropped frames instantly, also the flickering is visible here.

https://youtu.be/77fZfi0bILk

**Comment by dPing** [comment]
Wazerstar, you cannot expect CBR to stay within range when you specified 1QP min and 39max. Set both to -1 and retest or back to default at 18min and 51max. I was planning on disabling al qp with cbr and seeing where otwent nut I got busy tonight.

**Comment by wazerstar** [comment]
dPing what are you on about? In the older version it worked fine, also did a test with 18qp and max51, same result.

**Comment by kurdtkobain** [comment]
in enc-h264-simple.cpp around line 308-ish you never call `m_VideoEncoder->SetRateControlMethod((Plugin::AMD::H264RateControlMethod)obs_data_get_int(settings, AMF_VCE_H264_RATECONTROL));`

EDIT: Nevermind you do at line 302, but it does not get set looking at the logs the rate control method is set to CQP....
EDIT2: BTW Line 680 and 682 of amd-amf-h264.cpp say `<Plugin::AMD::H264VideoEncoder::GetPeakBitrate>` instead of `<Plugin::AMD::H264VideoEncoder::GetRateControlMethod>`

**Comment by Xaymar** [comment]
@wazerstar Please open up a new issue for this and submit the necessary information as shown here: https://github.com/Xaymar/OBS-AMD-Media-Framework/wiki/Reporting-Issues

**Comment by Xaymar** [comment]
@kurdtkobain Fixed the logging lines.

**Comment by kurdtkobain** [comment]
@Xaymar from what I was able to tell from my tests you have to call `SetRateControlMethod` after you call `SetRateControlSkipFrameEnabled` or the value of `SetRateControlMethod`  is not set in `m_VideoEncoder` for some odd reason?

EDIT: This was tested on my computer, an A8-7600 APU with a XFX Double Dissipation R7 360

**Comment by Xaymar** [comment]
@kurdtkobain The issue is actually here:

```
void Plugin::AMD::H264VideoEncoder::SetRateControlSkipFrameEnabled(bool enabled) {
    AMF_RESULT res = m_AMFEncoder->SetProperty(AMF_VIDEO_ENCODER_RATE_CONTROL_METHOD, enabled);
    if (res != AMF_OK) {
        ThrowExceptionWithAMFError("<Plugin::AMD::H264VideoEncoder::SetRateControlSkipFrameEnabled> Setting to %s failed with error %s (code %d).", enabled ? "Enabled" : "Disabled", res);
    }
    AMF_LOG_INFO("<Plugin::AMD::H264VideoEncoder::SetRateControlSkipFrameEnabled> Set to %s.", enabled ? "Enabled" : "Disabled");
}

bool Plugin::AMD::H264VideoEncoder::IsRateControlSkipFrameEnabled() {
    bool enabled;
    AMF_RESULT res = m_AMFEncoder->GetProperty(AMF_VIDEO_ENCODER_RATE_CONTROL_METHOD, &enabled);
    if (res != AMF_OK) {
        ThrowExceptionWithAMFError("<Plugin::AMD::H264VideoEncoder::IsRateControlSkipFrameEnabled> Retrieving Property failed with error %s (code %d).", res);
    }
    AMF_LOG_INFO("<Plugin::AMD::H264VideoEncoder::IsRateControlSkipFrameEnabled> Retrieved Property, Value is %s.", enabled ? "Enabled" : "Disabled");
    return enabled;
}
```

Notice the mistake? I certainly didn't until now.

**Comment by razorlikes** [comment]
So what does this mean? Will I be able to stream using this encoder?

**Comment by Xaymar** [comment]
Yes, once the patch is out.

**Comment by Xaymar** [comment]
Should be solved by 1.3.0pre3.
